pdns/syncres.cc

```diff
@@ -1238,6 +1238,10 @@ bool SyncRes::doCNAMECacheCheck(const DNSName &qname, const QType &qtype, vector
     return false;
   }
 
+  if (qtype == QType::DS || qtype == QType::DNSKEY) {
```

Comment:
I wonder if that check is not done too soon? That does skip just-in-time DNSSEC validation, for example, perhaps we should do it later as done in https://github.com/PowerDNS/pdns/commit/b0d6e4085102a4cff5105ba02cea91d0bbf7d937 ?